test/sql/test_compiler.py

```diff
@@ -1396,12 +1397,24 @@ def test_nested_conjunctions_short_circuit(self):
                 and_(
                     or_(
                         or_(t.c.x == 12),
-                        and_(or_(), or_(and_(t.c.x == 8)), and_()),
```

Comment:
**mike bayer** (zzzeek) wrote:

this is likely testing that empty or_() and and_() do what's expected so we should add boolean_empty here.